### PR TITLE
hal: armv8m: add reboot platformctl for nrf91

### PR DESCRIPTION
**Pull request by damianloew**
## Motivation and Context

Finalizing the nrf9160 basic port

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (nrf9160-dk).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/armv8m/nrf/91/nrf91.c [hunk 1]
```diff
@@ -47,13 +50,33 @@ enum { fpu_cpacr = 34, fpu_fpccr = 141, fpu_fpcar, fpu_fpdscr };
...
+	switch (data->type) {
+		case pctl_reboot:
+			if (data->action == pctl_set) {
+				if (data->reboot.magic == PCTL_REBOOT_MAGIC)
+					_interrupts_nvicSystemReset();
+			}
+			else if (data->action == pctl_get) {
+				data->reboot.reason = nrf91_common.resetFlags;
+				ret = 0;
+			}
```
Comment (lukileczo) [inline hal/armv8m/nrf/91/nrf91.c:73]: It's awful, but MISRA (rule 15.7)
```suggestion
			else {
				if (data->action == pctl_get) {
					data->reboot.reason = nrf91_common.resetFlags;
					ret = 0;
				}
			}
```
or keep `else if` and add empty `else`

File: hal/armv8m/nrf/91/nrf91.c [hunk 2]
```diff
@@ -47,13 +50,33 @@ enum { fpu_cpacr = 34, fpu_fpccr = 141, fpu_fpcar, fpu_fpdscr };
...
+			break;
+		default:
+			ret = -EINVAL;
```
[the remark this answers is not in the document]
Comment (damianloew) [reply hal/armv8m/nrf/91/nrf91.c:76]: Right, thanks for suggestions!

**Review by lukileczo — APPROVED** [review]
LGTM

File: hal/armv8m/nrf/91/nrf91.c [hunk 3]
```diff
@@ -47,13 +50,37 @@ enum { fpu_cpacr = 34, fpu_fpccr = 141, fpu_fpcar, fpu_fpdscr };
...
+				if (data->reboot.magic == PCTL_REBOOT_MAGIC) {
+					_interrupts_nvicSystemReset();
+				}
+			}
+			else {
+				if (data->action == pctl_get) {
+					data->reboot.reason = nrf91_common.resetFlags;
+					ret = 0;
+				}
+			}
```
Comment (agkaminski) [inline hal/armv8m/nrf/91/nrf91.c:76]: `ret` might be uninitialized in this function, no warn because nrf is not enabled on CI

Comment (damianloew) [reply hal/armv8m/nrf/91/nrf91.c:76]: Done, thanks!

Comment (agkaminski) [reply hal/armv8m/nrf/91/nrf91.c:76]: Now you return -EINVAL on pctl_get

Comment (damianloew) [reply hal/armv8m/nrf/91/nrf91.c:76]: What do you mean? It's set to 0 then

Comment (agkaminski) [reply hal/armv8m/nrf/91/nrf91.c:76]: Sorry, it's ok

**Review by agkaminski — APPROVED** [review]

**Pull request merged**